### DISCUSSION TITLE
Thematic Mapping

### DISCUSSION THREAD

**Discussion by stuartlynn** [maplibre/maplibre-rs#267] [Ideas]
Hey folks. I have been playing around building a crate for thematic mapping. The initial idea is that it would sit on top of mapbox-gl-js / maplibre-gl-js, kind of like a rust version of [deck.gl](http://deck.gl/). Made some progress with it but I have started wondering recently if it makes sense to combine those efforts with this project. As far as I can tell, what would be required is

- Adding non tiled data sources (geojson, csv, flatgeobuff, geoparquet etc)
- Defining new thematic LayerStyle types and adding these to the Style spec
- Writing new shaders where required for new visualization options or when we need to efficiently render large amounts of data

It would be amazing to have a single portable solution for base mapping and thematic mapping all in one but I am aware that this could be seen as scope creep for this project. If it's in scope I am happy to start thinking about how it could be integrated and start scoping out some tickets. Also happy to work towards implementing those features if they feel right for the project.

Thanks!

**Comment by maxammann** [comment]
(responded on element chat now)

https://matrix.to/#/!FJjNawYTQMCYZmCfbw:matrix.org/$LbaTcuLxQ9T7oFTKO1N-4GsFbD_A20f4I23pVCLYFuQ?via=matrix.org&via=chard.com&via=schinas.net

Feel free to discuss here or there.

> Having visualisations like [deck.gl](http://deck.gl/) in maplibre-rs is definitely something which is in-scope as a separate plugin.
> maplibre-rs got a plugin architecture (similar to bevy game engine), which means that it is possible to implement features like a 3D plot without touching the core parts of maplibre-rs
> 
> Now, I'm not sure how actually [deck.gl](http://deck.gl/) works from a technical point of view. I think this might also change with WebGL vs WebGPU.
> maplibre-gl-js and [deck.gl](http://deck.gl/) both render into the same framebuffer. But I'm not sure how they do it and if its a hack or if it is a clean solution.
> 
> So because maplibre-rs is based on WebGPU it might be difficult to make it work with maplibre-gl-js. Though not impossible, because currently maplibre-rs translates GPU commands to WebGL.
